pkg/api/api.go

```diff
@@ -217,11 +232,15 @@ func GetInstr(
 			}
 		})
 
-		return tracing.HTTPMiddleware(tracer, name, logger,
-			ins.NewHandler(name,
-				logMiddleware.HTTPMiddleware(name,
-					gziphandler.GzipHandler(
-						middleware.RequestID(hf),
+		return getTenantFromHeaders(
+			tracing.HTTPMiddleware(tracer, name, logger,
+				ins.NewHandler(name,
+					ins.InsertTenantIdentifier(
```

Comment:
I wonder what's the plan to propagate this header/metadata for further clients (e.g StoreAPI calls)